src/main/java/htsjdk/variant/vcf/AbstractVCFCodec.java

```diff
@@ -621,31 +618,50 @@ private static void parseSingleAltAllele(List<Allele> alleles, String alt, int l
     }
 
     public static boolean canDecodeFile(final String potentialInput, final String MAGIC_HEADER_LINE) {
+        InputStream fileInputStream = null;
         try {
-            Path path = IOUtil.getPath(potentialInput);
-            //isVCFStream closes the stream that's passed in
-            return isVCFStream(Files.newInputStream(path), MAGIC_HEADER_LINE) ||
-                    isVCFStream(new GZIPInputStream(Files.newInputStream(path)), MAGIC_HEADER_LINE) ||
-                    isVCFStream(new BlockCompressedInputStream(Files.newInputStream(path)), MAGIC_HEADER_LINE);
+            final Path path = IOUtil.getPath(potentialInput);
+            fileInputStream = Files.newInputStream(path);
+            BufferedInputStream bufferedInputStream = new BufferedInputStream(fileInputStream);
```

Comment:
Add a buffer size argument to the constructor to ensure that it can reset safely (`MAGIC_HEADER_LINE.length()`)

Reply:
I wouldn't want such a small buffer size because reading from the`GZIPInputStream` will read more than `MAGIC_HEADER_LINE.length()` bytes, since it will also read the gzip header. Changing this causes `Resetting to invalid mark` IO exceptions (which are then swallowed) when these guys try to reset after reading with the `GZIPInputStream`. Without this argument I was using the default `BufferedInputStream` buffer size, which is 8192. Is there a better buffer size to use?

Reply:
What about `Defaults.NON_ZERO_BUFFER_SIZE` and/or ensure enough buffer size for the magic string `Math.max(Defaults.NON_ZERO_BUFFER_SIXE, MAGIC_HEADER_LINE.length())`?

For the `GZIPInputStream` there is also a constructor with buffer size; doesn't it work in that case?